### PR TITLE
Adds option to EcalDeadCellTriggerPrimitiveFilter to consider kTPSaturated flag

### PR DESCRIPTION
**Pull request by kencall**
Ecal rechits have the flag, kTPSaturated, available. The filter has be updated and given a configurable option to consider this flag. The default configuration maintains the classical behavior.

Validation of the use of kTPSaturated is presented here: https://cms-cpt-software.web.cern.ch/cms-cpt-software/General/Validation/SVSuite/HCAL/2015D-EcalTP.pdf

Regards,
Ken Call

### CODE REVIEW

**Comment by slava77** [comment]
@cmsbuild please test

File: RecoMET/METFilters/plugins/EcalDeadCellTriggerPrimitiveFilter.cc [hunk 1]
```diff
@@ -158,6 +162,9 @@ class EcalDeadCellTriggerPrimitiveFilter : public edm::EDFilter {
 
   bool useTPmethod_, useHITmethod_;
 
+  bool exclusivelykTPSaturated_;
+  bool considerkTPSaturated_;
```
Comment (slava77) [inline RecoMET/METFilters/plugins/EcalDeadCellTriggerPrimitiveFilter.cc]: I'm not sure why these two are added: there are not used in the class methods.
Please clarify.

Comment (kencall) [inline RecoMET/METFilters/plugins/EcalDeadCellTriggerPrimitiveFilter.cc]: Hi Slava,

During development these names were used before the options useTTsum_ and usekTPSaturated_ were added. These two lines were left behind when the code was cleaned up in preparation for the PR. They can be removed, and I've updated the PR with them removed.

Regards,
Ken Call
________________________________

**Comment by slava77** [comment]
@cmsbuild please test

@kencall  thank you for the prompt cleanup

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/13429	1bf1b9f
- code changes are in line with the description
- jenkins tests of 302dbbb pass and comparisons with baseline show no differences
- jenkins tests of 1bf1b9f required just compilation, which passed OK; the errors are present in the IB (fastsim got broken)

**Comment by slava77** [comment]
@cmsbuild please test
maybe tests complete this time

**Comment by slava77** [comment]
@cmsbuild please test
looks like an IB with a fix to fastsim is out

File: RecoMET/METFilters/plugins/EcalDeadCellTriggerPrimitiveFilter.cc [hunk 2]
```diff
@@ -433,8 +440,13 @@ int EcalDeadCellTriggerPrimitiveFilter::setEvtRecHitstatus(const double &tpValCu
      if( !ebrechit->isRecovered() ) toDo = false;
 //     if( !ebrechit->checkFlag(EcalRecHit::kTowerRecovered) ) toDo = false;
 
+
+
      if( toDo ){
 
+	//If we considerkTPSaturated and a recHit has a flag set, we can immediately flag the event.
+        if(ebrechit->checkFlag(EcalRecHit::kTPSaturated) && usekTPSaturated_) return 1;
```
Comment (davidlange6) [inline RecoMET/METFilters/plugins/EcalDeadCellTriggerPrimitiveFilter.cc]: @kencall would be more efficient to first check useTPSaturated_ here and below. Something to consider for your next PR.

**Pull request merged**